src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,112 @@
 package core.basesyntax;
 
+
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private Node<K, V>[] table;
+    private int size;
 
-    @Override
-    public void put(K key, V value) {
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        size = 0;
+    }
 
+    public void put(K key, V value) {
+        int hash = (key == null) ? 0 : key.hashCode();
+        int index = (key == null) ? 0 : getIndex(hash, table.length);
+        if (table[index] == null) {
+            table[index] = new Node<>(key, value, hash);
+            size++;
+        } else {
+            if (table[index].value == value) {
+                return;
+            }
+            if (table[index].key == key) {
+                table[index].value = value;
+                return;
+            }
+            Node<K, V> current = table[index];
+            while (current.next != null) {
+                if ((key == null && current.key == null) || (key != null && key.equals(current.key))) {
+                    current.value = value;
+                    return;
+                }
+                current = current.next;
+            }
+            current.next = new Node<>(key, value, hash);
+            size++;
+        }
+        if (size > table.length * LOAD_FACTOR) {
+            resizeTable();
+        }
     }
 
-    @Override
     public V getValue(K key) {
+        int hash = (key == null) ? 0 : key.hashCode();
+        int index = (key == null) ? 0 : getIndex(hash, table.length);
+        Node<K, V> current = table[index];
+        while (current != null) {
+            if ((key == null && current.key == null) || (key != null && key.equals(current.key))) {
+                return current.value;
+            }
+            current = current.next;
+        }
         return null;
     }
 
-    @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getIndex(int hash, int tableLength) {
+        return (hash & Integer.MAX_VALUE) % tableLength;
+    }
+
+    private void resizeTable() {
```

Comment:
I think that it is better to use threshold in if statements upper. Maybe to make it class field, and calculate it every resize.